sentinel-extension/sentinel-datasource-redis/src/main/java/com/alibaba/csp/sentinel/datasource/redis/RedisDataSource.java

```diff
@@ -0,0 +1,82 @@
+package com.alibaba.csp.sentinel.datasource.redis;
+
+import com.alibaba.csp.sentinel.datasource.AbstractDataSource;
+import com.alibaba.csp.sentinel.datasource.ConfigParser;
+import com.alibaba.csp.sentinel.log.RecordLog;
+import io.lettuce.core.RedisClient;
+import io.lettuce.core.api.sync.RedisCommands;
+import io.lettuce.core.pubsub.RedisPubSubAdapter;
+import io.lettuce.core.pubsub.StatefulRedisPubSubConnection;
+import io.lettuce.core.pubsub.api.sync.RedisPubSubCommands;
+
+
+/**
+ * Redis data resource. when data publish in redis,this will update rule config in time
+ *
+ * @author moon tiger
+ */
+
+public class RedisDataSource<T> extends AbstractDataSource<String, T> {
+
+    private RedisClient redisClient = null;
+
+    private String ruleKey;
+
+    public RedisDataSource(RedisClient client, String ruleKey, String channel, ConfigParser<String, T> parser) {
```

Comment:
Here the constructor should not directly accept a `RedisClient` as users don't have to know the internal client here. Users can provide host and port, then we create the client internal. So even if we changed the Redis client library, users don't have to care about that.

The constructor can be like this:

```java
public RedisDataSource(String host, int port, String ruleKey, String channel, ConfigParser<String, T> parser)
```

Reply:
yes. it should be decoupled. but when build `RedisClient` with lettuce. it has different mode to build like this:
 
Standalone mode build: 
```java
RedisURI masterUri = RedisURI.Builder.redis("master-host", 6379).build();
```

Sentinel mode build: 
```java
RedisURI sentinelUri = RedisURI.Builder.sentinel("sentinel-host", 26379, "master-name").build();
```
case i build a own constructor to  compatible to all:

```java
MyRedisURI{host,port,password,RedisConnectionType,ClientResources}
```

Reply:
That's okay, you can implement a universal `RedisConfig` class including these attributes, then users can pass a `RedisConfig` to the constructor.